articles/techniques/testing-and-debugging.md

```diff
@@ -214,15 +213,80 @@ $$
 calling <xref:microsoft.quantum.diagnostics.dumpmachine> generates this output:
 
 ```
-Ids:    [1;0;]
-Wavefunction:
-0:      0.707107        0
-1:      0               0
-2:      -0.5            -0.5
-3:      0               0
+# wave function for qubits with ids (least to most significant): 0;1
+∣0❭:	 0.707107 +  0.000000 i	 == 	**********           [ 0.500000 ]     --- [  0.00000 rad ]
+∣1❭:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+∣2❭:	-0.500000 + -0.500000 i	 == 	**********           [ 0.500000 ]   /     [ -2.35619 rad ]
+∣3❭:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+```
+
+The first row provides a comment with the IDs of the corresponding qubits in their significant order.
+The rest of the rows describe the probability amplitude of measuring the basis state vector $\ket{n}$ in both, cartesian and polar, formats. In detail for the first row:
+
+* **`∣0❭:`** this rows correspond to the `0` state vector
+* **`0.707107 +  0.000000 i`**: the probability amplitude in cartesian format.
+* **` == `**: the `equal` sign seperates both equivalent representations.
+* **`**********  `**: A graphical representation of the magnitude, the number of `*` is proportionate to the probability of measuring this state vector.
+* **`[ 0.500000 ]`**: the numeric value of the magnitude
+* **`    ---`**: A graphical representation of the amplitude's phase (see below).
+* **`[ 0.0000 rad ]`**: the numeric value of the phase (in radians).
+
+Both the magnitude and the phase are displayed with a graphical representation. The magnitude representation is straight-forward: it shows a bar of `*`, the bigger the probability the bigger the bar will be. For the phase, we show the following symbols to represent the angle based on ranges:
+
+```
```

Comment:
This might be easier to read as a DFM table, perhaps?

Reply:
Maybe later.